src/timearray.jl

```diff
@@ -69,13 +69,67 @@ length(ata::AbstractTimeSeries) = length(ata.timestamp)
 
 size(ta::TimeArray, dim...) = size(ta.values, dim...)
 
-###### iterator protocol #########
+###### iterator protocol ########
 
 start(ta::TimeArray)   = 1
 next(ta::TimeArray, i) = ((ta.timestamp[i], ta.values[i, :]), i + 1)
 done(ta::TimeArray, i) = (i > length(ta))
 isempty(ta::TimeArray) = (length(ta) == 0)
 
+###### equal ####################
+
+"""
+    ==(x::TimeArray, y::TimeArray)
+
+If `true`, all fields of `x` and `y` should be equal,
+meaning that the two `TimeArray`s have the same values at the same points in time,
+the same colnames and the same metadata.
+
+Implies
+
+```julia
+x.timestamp == y.timestamp &&
+x.values    == y.values    &&
+x.colnames  == y.colnames  &&
+x.meta      == y.meta
+```
+"""
+@generated function ==(x::TimeArray{T,N}, y::TimeArray{S,M}) where {T,S,N,M}
```

Comment:
This shouldn't require a generated function.
```julia
Base.:(==)(x::TimeArray{T,N}, y::TimeArray{S,N}) where {T,S,N} =
    all(f->getfield(x, f) == getfield(y, f), fieldnames(TimeArray))
Base.:(==)(x::TimeArray{T,N}, y::TimeArray{S,M}) where {T,S,N,M} = false
```

Reply:
Wow, it's elegant.
TIL, `all` will do early stopping.

Reply:
and done in following commits